proxy/src/main/java/com/wavefront/agent/queueing/SQSQueueFactoryImpl.java

```diff
@@ -112,14 +121,13 @@ private String getOrCreateQueue(String queueName) {
     try {
       if (StringUtils.isBlank(queueUrl)) {
         CreateQueueRequest request = new CreateQueueRequest();
-        request.
-            addAttributesEntry(QueueAttributeName.MessageRetentionPeriod.toString(), "1209600").
-            addAttributesEntry(QueueAttributeName.ReceiveMessageWaitTimeSeconds.toString(), "20").
-            addAttributesEntry(QueueAttributeName.VisibilityTimeout.toString(), "60").
-            setQueueName(queueName);
+        request
+            .addAttributesEntry(QueueAttributeName.MessageRetentionPeriod.toString(), "1209600")
+            .addAttributesEntry(QueueAttributeName.ReceiveMessageWaitTimeSeconds.toString(), "20")
+            .addAttributesEntry(QueueAttributeName.VisibilityTimeout.toString(), "60")
+            .setQueueName(queueName);
         CreateQueueResult result = client.createQueue(request);
         queueUrl = result.getQueueUrl();
-        queues.put(queueName, queueUrl);
```

Comment:
Why are we removing this?

Reply:
That's done on the line 76, doing it here can cause concurrent execution.
` String queueUrl = queues.computeIfAbsent(queueName, x -> getOrCreateQueue(queueName));`

Reply:
Thanks, LGTM